CONTRIBUTING.md

```diff
@@ -401,41 +410,52 @@ let's also make sure to `test` it and perhaps create an `example notebook`.
 
 #### Formatting/Linting
 
-You can format and lint your changes with the following commands in the root directory:
+We run an assortment of linters: `black`, `ruff`, `mypy`.
 
-```bash
-make format; make lint
-```
+If you have installed pre-commit hooks in this repo, they should have taken care of the formatting and linting automatically.
 
-You can also make use of our pre-commit hooks by setting up git hook scripts:
+If -- for whatever reason -- you would like to do it manually, you can format and lint your changes with the following commands in the root directory:
 
 ```bash
-pre-commit install
+make format; make lint
 ```
 
-We run an assortment of linters: `black`, `ruff`, `mypy`.
+Under the hood, we still install pre-commit hooks for you, so that you don't have to do this manually next time.
 
 #### Testing
 
-For bigger changes, you'll want to create a unit test. Our tests are in the `tests` folder.
-We use `pytest` for unit testing. To run all unit tests, run the following in the root dir:
+If you modified or added code logic, **create test(s)**, because they help preventing other maintainers from accidentally breaking the nice things you added / re-introducing the bugs you fixed.
 
-```bash
-pytest tests
-```
+- In almost all cases, add **unit tests**.
+- If your change involves adding a new integration, also add **integration tests**. When doing so, please [mock away][mck] the remote system that you're integrating LlamaIndex with, so that when the remote system changes, LlamaIndex developers won't see test failures.
 
-or
+Reciprocally, you should **run existing tests** (from every package that you touched) before making a git commit, so that you can be sure you didn't break someone else's good work.
 
-```bash
+(By the way, when a test is run with the goal of detecting whether something broke in a new version of the codebase, it's referred to as a "[regression test][reg]". You'll also hear people say "the test _regressed_" as a more diplomatic way of saying "the test _failed_".)
+
+Our tests are stored in the `tests` folders under each package directory. We use the testing framework [pytest][pyt], so you can **just run `pytest` in each package you touched** to run all its tests.
+
+Just like with formatting and linting, if you prefer to do things the [make][mkf] way, run:
+
+```shell
 make test
```

Comment:
i guess `make test` still runs pants (which tbh, most people should just run pytest directly 😅)